specs/bytecode-proof.md

```diff
@@ -0,0 +1,57 @@
+# Bytecode Proof
+
+The bytecode proof helps the EVM proof by making the bytecode (identified by its code hash) easily accessible. Each byte in the bytecode is made accessible with its position in the bytecode. It is also annotated with a flag indicating if the byte is an opcode or is simply data for a previous PUSH instruction.
+
+## Circuit Layout
+
+| Column                | Description                                                        |
+| --------------------- | ------------------------------------------------------------------ |
+| `q_first`             | `1` on the first row, else `0`                                     |
+| `q_last`              | `1` on the last row, else `0`                                      |
+| `hash`                | The keccak hash of the bytecode                                    |
+| `index`               | The position of the byte in the bytecode                           |
+| `byte`                | The byte data for the current position                             |
+| `is_code`             | `1` if the byte is code, `0` if the byte is PUSH data              |
+| `push_data_left`      | The number of bytes that still need to follow for PUSH data        |
```

Comment:
I was a bit confused at first about `push_data_left` naming, I thought it's some info about how much push data are to the left (direction), it's totally clear though from the description, not sure why I though this :). However, perhaps you can add a couple of words like: how many bytes of data are still left behind this position (might be even named push_data_behind, but it's really not important, you can ignore me :)).

Reply:
Interesting way to read `left` here, did not even think about that. :) Changed the description a bit more to what you described.